Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift

```diff
@@ -0,0 +1,115 @@
+import Foundation
+import XcodeProj
+
+public protocol DefaultSettingsProviding {
+    func defaultProjectSettings(project: Project,
+                                buildConfiguration: BuildConfiguration) -> [String: Any]
+
+    func defaultTargetSettings(target: Target,
+                               buildConfiguration: BuildConfiguration) -> [String: Any]
+}
+
+public final class DefaultSettingsProvider: DefaultSettingsProviding {
+    private static let essentialProjectSettings: Set<String> = [
+        "ALWAYS_SEARCH_USER_PATHS",
+        "DEBUG_INFORMATION_FORMAT",
+        "ENABLE_NS_ASSERTIONS",
+        "ENABLE_TESTABILITY",
+        "GCC_DYNAMIC_NO_PIC",
+        "GCC_OPTIMIZATION_LEVEL",
+        "GCC_PREPROCESSOR_DEFINITIONS",
+        "MTL_ENABLE_DEBUG_INFO",
+        "ONLY_ACTIVE_ARCH",
+        "CLANG_ANALYZER_NONNULL",
+        "CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION",
+        "CLANG_CXX_LANGUAGE_STANDARD",
+        "CLANG_CXX_LIBRARY",
+        "CLANG_ENABLE_MODULES",
+        "CLANG_ENABLE_OBJC_ARC",
+        "CLANG_ENABLE_OBJC_WEAK",
+        "COPY_PHASE_STRIP",
+        "ENABLE_STRICT_OBJC_MSGSEND",
+        "GCC_C_LANGUAGE_STANDARD",
+        "GCC_NO_COMMON_BLOCKS",
+        "PRODUCT_NAME",
+    ]
+
+    private static let essentialTargetSettings: Set<String> = [
+        "SDKROOT",
+        "CODE_SIGN_IDENTITY",
+        "LD_RUNPATH_SEARCH_PATHS",
+        "VALIDATE_PRODUCT",
+        "SWIFT_OPTIMIZATION_LEVEL",
+        "SWIFT_ACTIVE_COMPILATION_CONDITIONS",
+        "CURRENT_PROJECT_VERSION",
+        "DEFINES_MODULE",
+        "DYLIB_COMPATIBILITY_VERSION",
+        "DYLIB_CURRENT_VERSION",
+        "DYLIB_INSTALL_NAME_BASE",
+        "INSTALL_PATH",
+        "PRODUCT_NAME",
+        "SKIP_INSTALL",
+        "VERSION_INFO_PREFIX",
+        "VERSIONING_SYSTEM",
+        "TARGETED_DEVICE_FAMILY",
+        "EXECUTABLE_PREFIX",
+        "COMBINE_HIDPI_IMAGES",
+        "ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES",
+        "WRAPPER_EXTENSION",
+    ]
+
+    private let settingsHelper = SettingsHelper()
```

Comment:
Same here, if the `DefaultSettingsProvider` methods are called more than once with different projects and targets, we might get unexpected results.

Reply:
Sure, I can move it to the methods. I've put it there to highlight what kind of "external" code is used by the class, also it would be easier to make it "injectable" in the future (if we think it helps).The class is stateless, the number of calls won't make any difference. 

I just wanted to avoid recreating it. It probably does not deserve a protocol at this point. It's an internal detail of that class, extracted to readability and testability of that code.

Reply:
> The class is stateless, the number of calls won't make any difference.

It's behaves as a stateless class because of how we use it, but nothing prevents us from using it differently. I'd aim for pure functions that hold no state outside the scope of the function.

Reply:
I wasn't sure what you mean by
> It's behaves as a stateless class because of how we use it

`SettingsHelpers` has no properties as it's a box class for helper functions, I created it to make it easier to test, to reduce responsibility of the `ConfigGenerator` and reuse the code. 

I felt that keeping it as ivar was the right choice but making it a local variable can work as well (I've updated the code)